core/block/editor/state/change.go

```diff
@@ -251,6 +251,7 @@ func (s *State) changeBlockDetailsSet(set *pb.ChangeDetailsSet) error {
 			Fields: make(map[string]*types.Value),
 		}
 	}
+	shortenDetailsToLimit(s.rootId, map[string]*types.Value{set.Key: set.Value})
```

Comment:
It's a little counterintuitive that Value is being shorted after calling `shortenDetailsToLimit`. Maybe extract the part of this function that handles the single value?

Reply:
Yeah, as we assume that each value should be limited to 64 kilobytes, pull out single-value logic to separate function